src/symbolize/gimli/libs_dl_iterate_phdr.rs

```diff
@@ -73,6 +84,19 @@ unsafe extern "C" fn callback(
             OsStr::from_bytes(unsafe { CStr::from_ptr(dlpi_name) }.to_bytes()).to_owned()
         }
     };
+    #[cfg(target_os = "android")]
+    let zip_offset: Option<u64> = {
+        // only check for ZIP-embedded file if we have data from /proc/self/maps
+        maps.as_ref().and_then(|maps| {
+            // check if file is embedded within a ZIP archive by searching for `!/`
+            super::extract_zip_path_android(&name).and_then(|_| {
+                // find MapsEntry matching library's base address and get its file offset
+                maps.iter()
+                    .find(|m| m.ip_matches(dlpi_addr as usize))
+                    .map(|m| m.offset())
```

Comment:
Note to self: reread the MapsEntry struct and its FromStr if you somehow develop another question about this.